dali/base/dasds.cpp

```diff
@@ -5947,6 +5948,18 @@ static int extNcompareFunc(CInterface * const *_itm1, CInterface * const *_itm2)
     return 1;
 }
 
+void CCovenSDSManager::checkEnvironment(IPropertyTree *oldEnv, IPropertyTree *newEnv)
+{
+    if (!areMatchingPTrees(oldEnv, newEnv))
+    {
+        StringBuffer filename("environment");
+        addFileTimestamp(filename);
+        filename.append(".xml");
+        PROGLOG("Detected environment change, backing up previous version to: %s", filename.str());
+        saveXML(filename, oldEnv, 0, XML_Format|XML_SortTags); // sort for easier diffing
```

Comment:
I had envisaged the old environments would be saved within Dali rather than as separate files on disk.